universe/archive.go

```diff
@@ -698,6 +698,13 @@ func (a *Archive) UniverseLeafKeys(ctx context.Context,
 func (a *Archive) MintingLeaves(ctx context.Context,
 	id Identifier) ([]Leaf, error) {
 
+	// Sanity check that the proof type is set to issuance. Only this proof
+	// type is valid for minting leaves.
+	if id.ProofType != ProofTypeIssuance {
```

Comment:
Surprised the itests aren't failing here. AFAICT, we still use this for normal transfer proofs as well: https://github.com/lightninglabs/taproot-assets/blob/1660f9ccd5d40717216a93fea16fa97491f7429f/rpcserver.go#L5399. 

So perhaps we just change the name of the method, and also the cal stack all the way down to the db?

Reply:
I believe that we have insufficient itest coverage. Method `MintingLeaves` is actually used to retrieve transfer proof leaves.

Our universe sync itest does not cover transfer proof syncing sufficiently, that's why my proof type check passed CI.

In the latest commits to this PR I've renamed method `MintingLeaves` to `FetchLeaves`. I'll put up an issue to improve itest coverage of transfer proof universe syncing.

Reply:
Issue: https://github.com/lightninglabs/taproot-assets/issues/1543